packages/react/src/components/SidePanel/SidePanel.jsx

```diff
@@ -0,0 +1,143 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+  Close16,
+  ChevronLeft16,
+  ChevronRight16,
+  OpenPanelLeft16,
+  OpenPanelRight16,
+} from '@carbon/icons-react';
+import classNames from 'classnames';
+
+import Button from '../Button';
+import { settings } from '../../constants/Settings';
+
+const { iotPrefix } = settings;
+
+const propTypes = {
+  direction: PropTypes.string,
+  open: PropTypes.bool,
+  slideOver: PropTypes.bool,
+  inline: PropTypes.bool,
+  slideIn: PropTypes.bool,
+  showCloseButton: PropTypes.bool,
+  showDrawer: PropTypes.bool,
+  title: PropTypes.string,
+  content: PropTypes.node,
+  onClose: PropTypes.func,
+  primaryButton: PropTypes.node,
+  secondaryButton: PropTypes.node,
+  icons: PropTypes.arrayOf(PropTypes.node),
+  testId: PropTypes.string,
+  condensed: PropTypes.bool,
+};
+const defaultProps = {
+  direction: 'end',
+  open: false,
+  slideOver: false,
+  inline: false,
+  slideIn: false,
+  showCloseButton: false,
+  showDrawer: false,
+  title: '',
+  content: '',
+  primaryButton: null,
+  secondaryButton: null,
+  testId: undefined,
+  icons: undefined,
+  condensed: false,
+  onClose: () => {},
+};
+
+const SidePanel = ({
+  open,
+  title,
+  slideOver,
+  inline,
+  slideIn,
+  content,
+  showCloseButton,
+  showDrawer,
+  onClose,
+  primaryButton,
+  secondaryButton,
+  direction,
+  icons,
+  testId,
+  condensed,
+}) => {
+  const getIcon = () => {
+    let icon;
+    if (slideOver && showCloseButton) {
+      icon = Close16;
+    }
+
+    if (inline && direction === 'start') {
+      icon = open ? ChevronLeft16 : OpenPanelLeft16;
+    }
+
+    if (inline && direction === 'end') {
+      icon = open ? ChevronRight16 : OpenPanelRight16;
+    }
+
+    return icon;
+  };
+
+  return (
+    <div
+      data-testid={testId}
+      className={classNames(`${iotPrefix}--side-panel`, {
+        [`${iotPrefix}--side-panel--drawer`]: showDrawer && !open,
+        [`${iotPrefix}--side-panel--inline`]: inline,
+        [`${iotPrefix}--side-panel--slide-over`]: slideOver,
+        [`${iotPrefix}--side-panel--slide-in`]: slideIn,
+        active: open,
```

Comment:
The class here should follow a BEM structure. 
```suggestion
        [`${iotPrefix}--side-panel--active`]: open,
```